File: ai-mano-faz-os-arquivos-principais/keep_alive.py
from __future__ import annotations

from threading import Thread

from flask import Flask, jsonify

app = Flask(__name__)


@app.get("/")
def home() -> tuple[dict[str, str], int]:
    return jsonify({"status": "online", "service": "discord-bot"}), 200


def _run_web_server(port: int) -> None:
    app.run(host="0.0.0.0", port=port, debug=False, use_reloader=False)


def keep_alive(port: int) -> Thread:
    thread = Thread(target=_run_web_server, args=(port,), daemon=True)
    thread.start()
    return thread

File: keep_alive.js
from flask import Flask
from threading import Thread

app = Flask('')

@app.route('/')
def home():
    return "recado importante: o kae é gay e sempre foi (online na porta 3000)"

def run():
    app.run(host='0.0.0.0', port=3000)

def keep_alive():
    t = Thread(target=run)
    t.start()
